app/economy/utils.py

```diff
@@ -27,23 +27,34 @@ class TransactionException(Exception):
     pass
 
 
-def convert_amount(from_amount, from_currency, to_currency):
+def convert_amount(from_amount, from_currency, to_currency, timestamp=None):
     """Convert the provided amount to another current.
 
     Args:
         from_amount (float): The amount to be converted.
         from_currency (str): The currency identifier to convert from.
         to_currency (str): The currency identifier to convert to.
+        timestamp (datetime): First available conversion rate after timestamp. Latest if None.
 
     Returns:
         float: The amount in to_currency.
 
     """
-    latest_conversion_rate = ConversionRate.objects.filter(
-        from_currency=from_currency,
-        to_currency=to_currency
+    conversion_rate = None
+
+    if timestamp:
+        conversion_rate = ConversionRate.objects.filter(
+            from_currency=from_currency,
+            to_currency=to_currency,
+            timestamp__gte=timestamp
+        ).order_by('-timestamp').last()
+    if not timestamp or not conversion_rate:  # fallback to latest if not found or timestamp not provided
```

Comment:
> or not conversion_rate

are we sure we want to do this?  im not sure convert_amount should just passively fallback to the most recent timestamp if conversion rates near that time are not found

Reply:
if `timestamp` is passed in and there's no data for that time range,  we should probably throw a `ConversationRateNotFoundException` -- and we'll have to make sure that's handled upstream too

Reply:
Done. I added ConversationRateNotFoundException, throwing it here and catching in value_in_usdt/value_in_usdt_now.